### ISSUE TITLE
OperationInterruptedException, ObjectDisposedException and NullReferenceException when CommandProcessor is scoped

### ISSUE THREAD

**Issue by dsquadcgi** [BrighterCommand/Brighter#1756]
#### TL;DR;

When CommandProcessor is registered as Scoped, several exceptions may be raised (all stacktraces are in the repo):
- OperationInterruptedException
- ObjectDisposedException
- NullReferenceException

When CommandProcessor is registered as Singleton (default), no exceptions are raised.

#### Details

In many of ours handlers, we inject gRPC clients that are registered with standard client factory, as scoped ([doc](https://docs.microsoft.com/en-us/aspnet/core/grpc/clientfactory?view=aspnetcore-5.0)). 

As CommandProcessor is singleton by default, the DI framework cannot resolve scoped gRPC clients. We could bypass this by manually creating a scope (something like [this](https://docs.microsoft.com/en-us/dotnet/core/extensions/scoped-service#rewrite-the-worker-class)), but we aren't so confident about it.

In such a situation, your documentation advises to explicitely set the CommandProcessor lifetime to Scoped. But when we do such, exceptions are raised.

### To Reproduce

Here is a repo to reproduce these issues: https://github.com/dsquadcgi/Brighter-NullReferenceException, with a simple WebApi and a xUnit WebApplicationFactory test.

I had to run several times the test to be able to reproduce, but from my experiments I can tell:
- `OperationInterruptedException` is raised often (more than 20 times per 5 000 API calls)
- `ObjectDisposedException` and `NullReferenceExceptionwas` were only thrown 1 time out of maybe 30 000 API calls ...

You have all the stacktraces in `exceptionRaised` folder.

### Exceptions (if any)
See `exceptionRaised` folder in repo.

### Further technical details
- Brighter version: 8.1.1399
- Include the output of `dotnet --info`: 

```
SDK .NET (reflétant tous les fichiers global.json) :
 Version:   5.0.300
 Commit:    2e0c8c940e

Environnement d'exécution :
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.300\

Host (useful for support):
  Version: 5.0.6
  Commit:  478b2f8c0e

.NET SDKs installed:
  3.1.100 [C:\Program Files\dotnet\sdk]
  5.0.300 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

Note that `NullReferenceException` also happens in our docker containers.

- The OS (Window, Linux, MacOs, etc): Windows 10, linux docker containers

Thanks a lot in advance!

**Comment by iancooper** [comment]
Hi @dsquadcgi thanks very much for the bug report, and the comprehensive way that it was filled out. We had a number of issues supporting scoped in v8, which we are fixing in v9. v9 is currently in pre-release, but I believe we are on the last of those issues - serviceactivator supporting scoped mesages - and then we will have a solution around this.

The solution comes from the fact that certain parts of our infrastructure our never per-request, by design, and we needed to cleanly separate those, so that we could support scopes.

We would not be able to backport this to v8 as it is a breaking change, hence why it is being released with v9

If you have questions, let me know, otherwise we will probably close this issue as resolved in v9.
